internal/core/command/device.go

```diff
@@ -15,43 +15,30 @@ package command
 
 import (
 	"context"
-	"fmt"
 	"net/http"
 
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/types"
-	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
 
 	"github.com/edgexfoundry/edgex-go/internal/core/command/errors"
-	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
+
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
 )
 
-func commandByDeviceID(deviceID string, commandID string, body string, queryParams string, isPutCommand bool, ctx context.Context) (string, int) {
+func commandByDeviceID(deviceID string, commandID string, body string, queryParams string, isPutCommand bool,
+	ctx context.Context) (string, error) {
 	d, err := mdc.Device(deviceID, ctx)
 	if err != nil {
-		LoggingClient.Error(err.Error())
-
-		chk, ok := err.(types.ErrServiceClient)
-		if ok {
-			return err.Error(), chk.StatusCode
-		} else {
-			return err.Error(), http.StatusInternalServerError
-		}
+		return "", err
 	}
 
 	if d.AdminState == contract.Locked {
-		LoggingClient.Error(d.Name + " is in admin locked state")
-		return errors.NewErrDeviceLocked(d.Name).Error(), http.StatusLocked
+		return "", errors.NewErrDeviceLocked(d.Name)
```

Comment:
Is there a way that we can move instances like this to the general error-checking approach? The goal of applying the general Error Handling is to provide a single place for error-checking. This will help facilitate that common error-handling approach we are aiming for in v2.

My first thought is that these utility methods (i.e. commandByDeviceId) are not too flexible in the sense that they are only re-used for testing sake. With that, more complexity has been introduced making it difficult to apply the common error-handling approach we've applied elsewhere.  Do you see the opportunity of refactoring this in such a way that we can cover scenarios like this which are not handled by the common handler? Possibly gutting this file of utility methods as a whole and find a way to reuse code while also applying common error handling.